neo/rawio/mearecrawio.py

```diff
@@ -36,52 +48,74 @@ class MEArecRawIO(BaseRawIO):
     extensions = ['h5']
     rawmode = 'one-file'
 
-    def __init__(self, filename=''):
+    def __init__(self, filename='', load_spiketrains=True, load_recordings=True):
         BaseRawIO.__init__(self)
         self.filename = filename
-
+        self.load_spiketrains = load_spiketrains
+        self.load_recordings = load_recordings
+        
     def _source_name(self):
         return self.filename
 
     def _parse_header(self):
+        load = ["channel_positions"] 
+        if self.load_recordings:
+            load.append("recordings")
+        if self.load_spiketrains:
+            load.append("spiketrains")
+        
         import MEArec as mr
         self._recgen = mr.load_recordings(recordings=self.filename, return_h5_objects=True,
                                           check_suffix=False,
-                                          load=['recordings', 'spiketrains', 'channel_positions'],
+                                          load=load,
                                           load_waveforms=False)
-        self._sampling_rate = self._recgen.info['recordings']['fs']
-        self._recordings = self._recgen.recordings
-        self._num_frames, self._num_channels = self._recordings.shape
-
-        signal_streams = np.array([('Signals', '0')], dtype=_signal_stream_dtype)
 
+        self.info_dict = deepcopy(self._recgen.info)
+        self.channel_positions = self._recgen.channel_positions
+        if self.load_recordings:
+            self._recordings = self._recgen.recordings
+        if self.load_spiketrains:
+            self._spiketrains = self._recgen.spiketrains
+
+        self._sampling_rate = self.info_dict['recordings']['fs']
+        self.duration_seconds = self.info_dict["recordings"]["duration"]
+        self._num_frames = int(self._sampling_rate * self.duration_seconds)
+        self._num_channels = self.channel_positions.shape[0]
+        self._dtype = self.info_dict["recordings"]["dtype"]
+        
+        signals = [('Signals', '0')] 
+        signal_streams = np.array(signals, dtype=_signal_stream_dtype)
+
+        
         sig_channels = []
         for c in range(self._num_channels):
             ch_name = 'ch{}'.format(c)
             chan_id = str(c + 1)
             sr = self._sampling_rate  # Hz
-            dtype = self._recordings.dtype
+            dtype = self._dtype
             units = 'uV'
             gain = 1.
             offset = 0.
             stream_id = '0'
             sig_channels.append((ch_name, chan_id, sr, dtype, units, gain, offset, stream_id))
+    
         sig_channels = np.array(sig_channels, dtype=_signal_channel_dtype)
```

Comment:
With load_recording=False this should be empty.

Reply:
This is empty now.